### PR TITLE
Feat auth tests

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
GALI-722

Description of changes:
Sign In/Out, Auth resource API tests

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/integration-tests/support/setup.js [hunk 1]
```diff
@@ -137,12 +137,15 @@ class Setup {
...
-    projectId = [this.defaults.project.id],
-  } = {}) {
+  async createUserSession(
+    {
+      userRole = 'internal-guest',
+      username = this.gen.username(),
+      password = this.gen.password(),
+      projectId = [this.defaults.project.id],
+    } = {},
+    { includeCreds = false } = {},
```
Comment (hatimkhan) [inline main/integration-tests/support/setup.js]: Per our Slack discussion, we can take a different approach to this.

Comment (SanketD92) [reply main/integration-tests/support/setup.js]: Done

File: main/integration-tests/__test__/api-tests/authentication/create-auth-id-token.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Creating an authentication token upon user login', () => {
+    it('should fail if empty request body is passed', async () => {
```
Comment (hatimkhan) [inline main/integration-tests/__test__/api-tests/authentication/create-auth-id-token.test.js:33]: Would you be able to add one more test where you pass random passwords instead of just an empty body?

Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/authentication/create-auth-id-token.test.js:33]: And another one where you pass random usernames and passwords

Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/authentication/create-auth-id-token.test.js:33]: Sure thing. Done.

File: main/integration-tests/__test__/api-tests/authentication/get-auth-provider-configs.test.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      });
+    });
+
+    it('should list auth provider configs', async () => {
+      const admin2Session = await setup.createAdminSession();
+      const response = await admin2Session.resources.authentication.configs().get();
+
+      await expect(response).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
```
Comment (hatimkhan) [inline main/integration-tests/__test__/api-tests/authentication/get-auth-provider-configs.test.js]: If the output shape is 100% to match, can you use expect.toStrictEqual()?

Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/authentication/get-auth-provider-configs.test.js]: If you put this part in a file say 'expected-response.js' and then require it in the test, it help readability.  This is just a thought.

Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/authentication/get-auth-provider-configs.test.js]: The output also contains dateTime elements, due to which objectContaining seemed to fit. However `toEqual` can be safely replaced with `toStrictEqual`.

Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/authentication/get-auth-provider-configs.test.js]: Not changing this, per our conversation.

File: main/integration-tests/__test__/api-tests/authentication/get-auth-provider-types.test.js [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should list auth provider types', async () => {
+      const admin2Session = await setup.createAdminSession();
+
+      await expect(admin2Session.resources.authentication.types().get()).resolves.toEqual(
+        expect.arrayContaining([
+          {
```
Comment (hatimkhan) [inline main/integration-tests/__test__/api-tests/authentication/get-auth-provider-types.test.js]: Same as other comment, what do you think of taking this into another file, the file returns a function that you can call and pass it whatever parameters you need, it should return this json object.

Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/authentication/get-auth-provider-types.test.js]: These are the only two built-in provider types that will always be returned for this API call. The function would have to return this list, regardless of the parameters, unless more provider types are added in the future. 

Let me know if I'm misunderstanding your comment.

Comment (SanketD92) [reply main/integration-tests/__test__/api-tests/authentication/get-auth-provider-types.test.js]: Moving the large JSON bodies in different files, per discussion.

File: main/integration-tests/support/resources/authentication/authentication-provider-types.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      type: 'authProviderTypes',
+      parent,
+    });
+    this.api = `${parent.api}/provider/types`;
+
+    if (_.isEmpty(parent)) throw Error('A parent resource was not provided to resource type [providerTypes]');
+  }
+
+  // ************************ Helpers methods ************************
+  getDefaultTypes() {
```
Comment (hatimkhan) [inline main/integration-tests/support/resources/authentication/authentication-provider-types.js]: minor.  Since we have .defaults(),  maybe it will make sense to drop 'get'  and just use  .defaultTypes()

File: main/integration-tests/support/resources/authentication/authentication-id-tokens.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    this.api = `${parent.api}/id-tokens`;
+  }
+
+  idToken(username) {
+    return new AuthenticationIdToken({ clientSession: this.clientSession, id: username, parent: this });
+  }
+
+  async request(body = {}, params = {}) {
+    return this.doCall(async () => this.axiosClient.post(this.api, body, { params }));
+  }
```
Comment (hatimkhan) [inline main/integration-tests/support/resources/authentication/authentication-id-tokens.js]: Did you face any issues by simply calling 'create()', otherwise, I would recommend using the create() method and not needing this helper method.  If you use the 'create()' method, the default implementation of create will add a cleanup task automatically so that the token will be cleaned after the session.

**Review by hatimkhan — DISMISSED** [review]

**Review by hatimkhan — APPROVED** [review]
